core/src/main/scala/org/apache/spark/scheduler/StageInfo.scala

```diff
@@ -36,7 +36,7 @@ class StageInfo(
     val rddInfos: Seq[RDDInfo],
     val parentIds: Seq[Int],
     val details: String,
-    val taskMetrics: TaskMetrics = new TaskMetrics,
+    val taskMetrics: TaskMetrics = null,
```

Comment:
creating `TaskMetrics` is not that cheap(register and un-register accumulators), so using null here as we won't call it when using default value.